### PR TITLE
Resolve #672, #715, #721

### LINKED ISSUE #672: Improve Ledger wallet connect process
- Make changes to Ledger wallet connect process in WalletSelectionComponent as well as Received WalletSelectionComponent.
- Instead of asking users to connect Ledger device **AND** navigate to the selected crypto app on Ledger, users can now first connect the device and later be guided to select the crypto app on their device, like what users would expect from Ledger Live.
- Make changes to some other components as well to accommodate this change.

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/actions/walletActions.js [hunk 1]
```diff
@@ -78,16 +79,64 @@ function onMetamaskAccountsChanged (accounts: Array<string>, cryptoType: string)
...
+    type: 'CHECK_LEDGER_APP_CONNECTION',
+    payload: async () => {
+      function sleep (time) {
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            resolve()
+          }, time)
+        })
+      }
+      let ledger = new LedgerNanoS()
```
Comment (timothywangdev) [inline src/actions/walletActions.js]: check validity of cryptoType first, if not valid, throw error

Comment (timothywangdev) [reply src/actions/walletActions.js]: use wallets.cryptoInWallet()

File: src/actions/walletActions.js [hunk 2]
```diff
@@ -78,16 +79,64 @@ function onMetamaskAccountsChanged (accounts: Array<string>, cryptoType: string)
...
+      while (true) {
+        try {
+          if (cryptoType === 'ethereum') {
```
Comment (timothywangdev) [inline src/actions/walletActions.js]: need to consider dai

File: src/containers/ReceiveWalletSelectionContainer.jsx [hunk 3]
```diff
@@ -109,10 +130,12 @@ class ReceiveWalletSelectionContainer extends Component {
 const checkWalletConnectionSelector = createLoadingSelector([
   'CHECK_CLOUD_WALLET_CONNECTION',
   'CHECK_METAMASK_CONNECTION',
-  'CHECK_LEDGER_NANOS_CONNECTION'
+  'GET_LEDGER_WALLET_DATA'
```
Comment (timothywangdev) [inline src/containers/ReceiveWalletSelectionContainer.jsx]: are we missing "CHECK_LEDGER_DEVICE_CONNECTION"

File: src/reducers/walletReducer.js [hunk 4]
```diff
@@ -70,14 +71,18 @@ export default function (state = initState, action) {
...
     case 'UPDATE_METAMASK_ACCOUNTS_FULFILLED':
-    case 'CHECK_LEDGER_NANOS_CONNECTION_FULFILLED':
+    case 'GET_LEDGER_WALLET_DATA_FULFILLED':
     case 'CHECK_CLOUD_WALLET_CONNECTION_FULFILLED':
     case 'CREATE_CLOUD_WALLET_FULFILLED':
     case 'GET_CLOUD_WALLET_FULFILLED':
       return updateWalletState(state, action.payload, { connected: true })
+    case 'CHECK_LEDGER_DEVICE_CONNECTION_FULFILLED':
+      return updateWalletState(state, [{ walletType: 'ledger' }], { connected: true })
     case 'CHECK_LEDGER_NANOS_CONNECTION_PENDING':
```
Comment (timothywangdev) [inline src/reducers/walletReducer.js]: do we still have these two actions?

File: src/containers/ReceiveWalletSelectionContainer.jsx (hunk 3 above)
Comment (InVincible2016) [reply src/containers/ReceiveWalletSelectionContainer.jsx]: No, CHECK_LEDGER_DEVICE_CONNECTION has it would create selector

File: src/wallet.js [hunk 5]
```diff
@@ -9,14 +9,17 @@ import { isMobile } from 'react-device-detect'
...
 
 export const walletCryptoSupports = {
-  'drive': [{ cryptoType: 'ethereum', disabled: false },
+  drive: [
+    { cryptoType: 'ethereum', disabled: false },
     { cryptoType: 'dai', disabled: false },
-    { cryptoType: 'bitcoin', disabled: false }],
-  'metamask': [{ cryptoType: 'ethereum', disabled: false },
-    { cryptoType: 'dai', disabled: false }],
-  'ledger': [{ cryptoType: 'ethereum', disabled: false },
```
Comment (timothywangdev) [inline src/wallet.js]: this is causing app to crash

Comment (timothywangdev) [reply src/wallet.js]: cryptoInWallet always returns false

File: src/actions/walletActions.js (hunk 1 above)
Comment (timothywangdev) [reply src/actions/walletActions.js]:
```
Invalid cryptoType ${cryptoType} for Ledger
```

Comment (timothywangdev) [reply src/actions/walletActions.js]: Also make cryptoInWallet accept string instead of an object

**Pull request merged**